### PR TITLE
fix git clone '.git' bug

### PR DESCRIPTION
**Pull request by lingfengchencn**
# What does this PR do?

fix git clone '.git' bug

Optimize installation directory to tmp and SLAM-LLM to /workspace/SLAM-LLM

Fixes # (issue)

### CODE REVIEW

**Comment by ZhikangNiu** [comment]
I don't think these modifications are necessary. Here are my points @lingfengchencn  @ddlBoJack :
- it's unnecessary add `cd ..`, because we have set WORKDIR
- It's more flexible install related python package in workspace. If the user want to change some source code in other repo, they can change them in container directly. This is why we use editable installation
- git support with/without .git suffix

File: Dockerfile [hunk 1]
```diff
@@ -16,26 +16,30 @@ RUN set -x \
...
-WORKDIR /workspace
+WORKDIR /tmp
 
-RUN git clone https://github.com/huggingface/transformers.git \
+RUN git clone https://github.com/huggingface/transformers \
     && cd transformers \
     && git checkout tags/v4.35.2 \
-    && pip install --no-cache-dir -e .
+    && pip install --no-cache-dir -e . \
+    && cd ..
```
Comment (ZhikangNiu) [inline Dockerfile:25]: unnecessary add `cd ..` and delete `.git`

File: Dockerfile [hunk 2]
```diff
@@ -16,26 +16,30 @@ RUN set -x \
 RUN pip install --no-cache-dir packaging editdistance gpustat wandb einops debugpy tqdm soundfile matplotlib scipy sentencepiece pandas \
     && pip install --no-cache-dir torchaudio==2.1.0 --index-url https://download.pytorch.org/whl/cu118
 
-WORKDIR /workspace
+WORKDIR /tmp
```
Comment (ZhikangNiu) [inline Dockerfile:19]: If we clone them in /tmp, why we need install them with editable?

**Comment by lingfengchencn** [comment]
OK， just close it.

**Pull request closed without merging**